drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3442,25 +3467,57 @@ static int rswitch_ndev_create(struct rswitch_private *priv, int index)
 	ndev->features = NETIF_F_RXCSUM;
 	ndev->hw_features = NETIF_F_RXCSUM;
 	ndev->base_addr = (unsigned long)rdev->addr;
-	snprintf(ndev->name, IFNAMSIZ, "tsn%d", index);
+	if (!rmon_dev) {
+		snprintf(ndev->name, IFNAMSIZ, "tsn%d", index);
+		ndev->ethtool_ops = &rswitch_ethtool_ops;
+		rswitch_set_mac_address(rdev);
+		rdev->mondev = false;
+	} else {
+		snprintf(ndev->name, IFNAMSIZ, "rmon%d", index);
+		eth_hw_addr_random(ndev);
+		rdev->mondev = true;
+		rdev->rmon_idx = index;
+	}
 	ndev->netdev_ops = &rswitch_netdev_ops;
-	ndev->ethtool_ops = &rswitch_ethtool_ops;
 
 	netif_napi_add(ndev, &rdev->napi, rswitch_poll, 64);
 
-	rswitch_set_mac_address(rdev);
 
 	/* FIXME: it seems S4 VPF has FWPBFCSDC0/1 only so that we cannot set
 	 * CSD = 1 (rx_default_chain->index = 1) for FWPBFCS03. So, use index = 0
 	 * for the RX.
 	 */
-	err = rswitch_rxdmac_init(ndev, priv, -1);
-	if (err < 0)
-		goto out_rxdmac;
+	if (!rmon_dev) {
+		err = rswitch_rxdmac_init(ndev, priv, -1);
+		if (err < 0)
+			goto out_rxdmac;
 
-	err = rswitch_txdmac_init(ndev, priv, -1);
-	if (err < 0)
-		goto out_txdmac;
+		err = rswitch_txdmac_init(ndev, priv, -1);
+		if (err < 0)
+			goto out_txdmac;
+	} else {
+		/* All rmon devices use the same chains because
+		 * CPU mirroring can mirror traffic only to one
+		 * sub-destination. The traffic will be forwarded
+		 * to appropriate netdevs in rswitch_rx function
+		 * according to source lock vector stored in info1.
+		 */
+		if (!priv->mon_rx_chain || !priv->mon_tx_chain) {
+			err = rswitch_rxdmac_init(ndev, priv, -1);
+			if (err < 0)
+				goto out_rxdmac;
+
+			err = rswitch_txdmac_init(ndev, priv, -1);
+			if (err < 0)
+				goto out_txdmac;
+
+			priv->mon_rx_chain = rdev->rx_default_chain;
```

Comment:
Nit: this common part can be executed unconditionally after the previous check

Reply:
There are no common parts, so I leave this as it was before.